plugin-volume/volumepopup.cpp

```diff
@@ -197,8 +208,23 @@ void VolumePopup::openAt(QPoint pos, Qt::Corner anchor)
 
 void VolumePopup::handleWheelEvent(QWheelEvent *event)
 {
-    m_volumeSlider->setSliderPosition(m_volumeSlider->sliderPosition()
-            + (event->angleDelta().y() / QWheelEvent::DefaultDeltasPerStep * m_volumeSlider->singleStep()));
+    static int _delta = 0; // for high-resolution mice and touchpad scrolling
+
+    int delta = event->angleDelta().y();
+    if ((_delta ^ delta) < 0)
+        _delta = delta; // the wheel direction is reversed
```

Comment:
Why this case ? Even if the user changes the wheel direction the previous "fraction" of the stroke has to be considered...
As counter example, if you in a normal mouse  each tick is emitted every 15°, and you rotate for +14° up and -14° down you don't have any effect. However with your code an hires mouse you have a tick because +14 - -14 = 28 > 15°.

Reply:
→ https://github.com/lxqt/lxqt-panel/pull/1856#issuecomment-1381367296 → 3

I'm afraid, you misread the code.